### PR TITLE
Dev-4556 Contract Grant Activity Chart Potential Award Amount Line

### PR DESCRIPTION
**Pull request by jonhill13**
**High level description:**

Implement the Potential Award Amount Line for the Contract Grants Activity Chart.

**Technical details:**

> • refactor vertical line component into svg line

> • confirmed with design no text.

**JIRA Ticket:**
[DEV-4556](https://federal-spending-transparency.atlassian.net/browse/DEV-4556)

**Mockup:**
https://bahdigital.invisionapp.com/share/7HIADD8Q53D#/screens/296034817

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR
- [x] Verified cross-browser compatibility
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension) Test with award, 66945037, in dev and local.  A decrease was viewed from 60 errors in dev to 58 errors locally.
- [N/A] Added Unit Tests for methods in Container Components, reducers, and helper functions `if applicable`
- [x] All componentWillReceiveProps, componentWillMount, and componentWillUpdate in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3339](https://federal-spending-transparency.atlassian.net/browse/DEV-3339)
- [N/A] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [N/A] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [N/A] Design review complete
- [N/A] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

File: src/js/components/award/shared/activity/ContractGrantActivityChartCircles.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    const [circleData, setCircleData] = useState([]);
+    // sets circle data - hook - on mount and when transactions change
+    useEffect(() => {
+        if (xScale && yScale) {
+            const circles = transactions
+                .map((data, i) => ({
+                    key: `${data.federal_action_obligation}${i}`,
+                    description: `A circle representing the transaction date, ${data.action_date.format('MM-DD-YYYY')} and running total obligation of ${formatMoney(data.running_obligation_total)}`,
+                    className: 'transaction-date-circle',
+                    cx: xScale(data.action_date.valueOf()) + padding.left,
```
Comment (MaxwellKendall) [inline src/js/components/award/shared/activity/ContractGrantActivityChartCircles.jsx:37]: why padding.left? Should the prop be just `paddingLeft`?

File: src/js/components/award/shared/activity/ContractGrantActivityChartCircles.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                    cy: (height - yScale(data.running_obligation_total)),
+                    r: 1.5
```
Comment (MaxwellKendall) [inline src/js/components/award/shared/activity/ContractGrantActivityChartCircles.jsx:39]: What is this? Maybe give a better name? Obviously its the same as the `circle` dom element, but might be helpful to give a different name here.

File: src/js/components/sharedComponents/SVGLine.jsx [hunk 3]
```diff
@@ -8,25 +8,30 @@ import PropTypes from 'prop-types';
...
+    position: PropTypes.number, // value of position
     showTextPosition: PropTypes.string, // show text left, right, and top are valid
     textY: PropTypes.number, // show text at this height
     description: PropTypes.string,
     adjustmentX: PropTypes.number, // adjust x for padding
     textClassname: PropTypes.string,
-    lineClassname: PropTypes.string
+    lineClassname: PropTypes.string,
+    noText: PropTypes.bool, // do not show text,
+    isHorizontal: PropTypes.bool,
```
Comment (MaxwellKendall) [inline src/js/components/sharedComponents/SVGLine.jsx:30]: Like the tooltip, we should use an enum to specify this

File: src/js/components/award/shared/activity/ContractGrantActivityChartCircles.jsx (hunk 2 above)
Comment (jonhill13) [reply src/js/components/award/shared/activity/ContractGrantActivityChartCircles.jsx:39]: r is radius.
https://developer.mozilla.org/en-US/docs/Web/SVG/Element/circle

File: src/js/components/award/shared/activity/ContractGrantActivityChartCircles.jsx (hunk 1 above)
Comment (jonhill13) [reply src/js/components/award/shared/activity/ContractGrantActivityChartCircles.jsx:37]: We treat padding as below for most charts.
```
{ left: 45, bottom: 30 }
```

File: src/js/components/sharedComponents/SVGLine.jsx (hunk 3 above)
Comment (jonhill13) [reply src/js/components/sharedComponents/SVGLine.jsx:30]: I could see that if we are drawing things other that horizontal or vertical but when there are two options is it not convention to use bool?

**Pull request merged**